parameterspace/configspace_utils.py

```diff
@@ -0,0 +1,130 @@
+                    " or ".join([f"{parent} == {repr(v)}" for v in cond["values"]])
+                )
+            elif cond["type"] == "EQ":
+                function_texts.append(f"{parent} == {repr(cond['value'])}")
+            else:
+                raise NotImplementedError(f"Unsupported condition type {cond['type']}")
+
+    if not varnames:
+        return condition
+
+    function_text = " and ".join(function_texts)
+    verify_lambda(variables=varnames, body=function_text)
+    # pylint: disable=eval-used
+    condition_function = eval(f"lambda {', '.join(varnames)}: {function_text}")
+    # pylint: enable=eval-used
+
+    condition.function_texts.append(function_text)
+    condition.varnames.append(varnames)
+    condition.all_varnames |= set(varnames)
+    condition.functions.append(condition_function)
+
+    return condition
+
+
+def parameterspace_from_configspace_dict(configspace_dict: dict) -> ps.ParameterSpace:
+    space = ps.ParameterSpace()
+
+    for param_dict in configspace_dict["hyperparameters"]:
+        param_name = _escape_parameter_name(param_dict["name"])
+        condition = _get_condition(configspace_dict["conditions"], param_dict["name"])
+        if param_dict["type"] == "uniform_int":
+            space._parameters[param_name] = {
+                "parameter": ps.IntegerParameter(
+                    name=param_name,
+                    bounds=(param_dict["lower"], param_dict["upper"]),
+                    transformation="log" if param_dict["log"] else None,
+                ),
+                "condition": condition,
+            }
+
+        elif param_dict["type"] == "categorical":
+            space._parameters[param_name] = {
+                "parameter": ps.CategoricalParameter(
+                    name=param_name,
+                    values=param_dict["choices"],
+                    prior=param_dict.get("weights", None),
+                ),
+                "condition": condition,
+            }
+
+        elif param_dict["type"] == "constant":
+            space._parameters[param_name] = {
+                "parameter": ps.CategoricalParameter(
+                    name=param_name,
+                    values=[param_dict["value"]],
+                ),
+                "condition": condition,
+            }
+            space.fix(**{param_name: param_dict["value"]})
+
+        elif param_dict["type"] == "normal_float":
+            lower_bound = param_dict["mu"] - 4 * param_dict["sigma"]
+            upper_bound = param_dict["mu"] + 4 * param_dict["sigma"]
+            if param_dict["log"]:
+                lower_bound = max(lower_bound, 1e-24)
+
+            space._parameters[param_name] = {
+                "parameter": ps.ContinuousParameter(
+                    name=param_name,
+                    bounds=(lower_bound, upper_bound),
+                    prior=ps.priors.TruncatedNormal(
+                        mean=param_dict["mu"], std=param_dict["sigma"]
```

Comment:
If I am not mistaken, `ps.priors.TruncatedNormal` expects the mean and standard deviation in the normalized space. So given the bounds from above, I think it should be:
```suggestion
                        mean=0.5, std=1./8.
```